fedimint-client-rpc/src/lib.rs

```diff
@@ -196,13 +198,21 @@ impl RpcGlobalState {
         let federation_secret = self.derive_federation_secret(&mnemonic, &federation_id);
 
         let builder = Self::client_builder(client_db).await?;
-        let client = Arc::new(
-            builder
-                .preview(&invite_code)
-                .await?
-                .join(RootSecret::StandardDoubleDerive(federation_secret))
-                .await?,
-        );
+        let preview = builder.preview(&invite_code).await?;
+
+        let client = if recover {
+            Arc::new(
+                preview
```

Comment:
you shouldn't expose recover: bool, you should download the backup from server. and if server does have the backup, you have to force recovery. otherwise you get nonce reuse with client.